src/api/profile.ts

```diff
@@ -38,3 +38,37 @@ export async function deleteProfile(opts?: AxiosRequestConfig) {
 
   return deleteOK;
 }
+
+/**
+ * Function to get customer account ID based on phone number
+ *
+ * @param phoneNumber phone number with prefix
+ *
+ * @returns customer account ID as string if request is success
+ *
+ * Will return one the following errors if request unsuccessful:
+ * -  Error 404 with object {"error": "cannot find user from phone number: \"{phoneNumber}\""}
+ * -  Error 400 with object {"error": "Invalid phone number as per country code"}
+ * -  other than those 2 errors mentioned earlier, returns undefined.
+ */
+export const getCustomerAccountId = async (phoneNumber: string) => {
+  return await client
+    .post(
+      `${profileEndpoint}/search`,
+      {phoneNumber: phoneNumber},
+      {
+        authWithIdToken: true,
+        skipErrorLogging: (error) =>
+          error.response?.status === 404 || error.response?.status === 400,
+      },
+    )
+    .then((response) => response.data.customerAccountId)
+    .catch((error) => {
+      const metadata = getAxiosErrorMetadata(error);
+      const responseStatus = metadata.responseStatus;
+      if (responseStatus === 404 || responseStatus === 400) {
+        return metadata.responseData && JSON.parse(metadata.responseData);
+      }
+      return undefined;
```

Comment:
Can we instead return undefined when no phone number was found? That is more intuitive to me than returning an error when phone number not found, and returning undefined if unknown error occured.